### PR TITLE
Add video-trimmer: a light video cutter in Rust

### PR DESCRIPTION
**Pull request by Kiri2002**
light weight video cutter written in Rust, included in gnome-circles
carry from [AUR](https://aur.archlinux.org/packages/video-trimmer)

### CODE REVIEW

**Comment by Kiri2002** [comment]
这个包，简单的剪视频很快很好用。因为用Rust写的，今天从AUR下载，编译好久。希望能打到CN仓库，后人直接下二进制，造福人类

File: archlinuxcn/video-trimmer/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  meson compile -C build
+}
+
+check() {
+  meson test -C build --print-errorlogs
+}
+
+package() {
+  meson install -C build --destdir "$pkgdir"
+}
```
Comment (Zenvie) [inline archlinuxcn/video-trimmer/PKGBUILD:36]:
```
video-trimmer W: ELF file ('usr/bin/video-trimmer') lacks GNU_PROPERTY_X86_FEATURE_1_SHSTK.
video-trimmer W: Unused shared library '/usr/lib/libpango-1.0.so.0' by file ('usr/bin/video-trimmer')
video-trimmer W: Unused shared library '/usr/lib/libgdk_pixbuf-2.0.so.0' by file ('usr/bin/video-trimmer')
video-trimmer W: Unused shared library '/usr/lib/libcairo.so.2' by file ('usr/bin/video-trimmer')
```

File: archlinuxcn/video-trimmer/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# Contributor: Mark Wagie <mark dot wagie at proton dot me>
+pkgname=video-trimmer
+pkgver=0.8.2
+pkgrel=1
+pkgdesc="Trim videos quickly"
+arch=('x86_64')
+url="https://apps.gnome.org/VideoTrimmer"
+license=('GPL3')
+depends=('ffmpeg' 'gst-plugins-good' 'libadwaita')
+makedepends=('blueprint-compiler' 'cargo' 'git' 'meson')
```
Comment (Zenvie) [inline archlinuxcn/video-trimmer/PKGBUILD:11]:
```
video-trimmer W: Dependency hicolor-icon-theme detected and implicitly satisfied (needed for hicolor theme hierarchy)
video-trimmer W: Dependency dconf detected and implicitly satisfied (needed for glib schemas)
video-trimmer W: Dependency glib2 detected and implicitly satisfied (libraries ['usr/lib/libglib-2.0.so.0', 'usr/lib/libgio-2.0.so.0', 'usr/lib/libgobject-2.0.so.0'] needed in files ['usr/bin/video-trimmer'])
video-trimmer W: Dependency gtk4 detected and implicitly satisfied (libraries ['usr/lib/libgtk-4.so.1'] needed in files ['usr/bin/video-trimmer'])
video-trimmer W: Dependency glibc detected and implicitly satisfied (libraries ['usr/lib/ld-linux-x86-64.so.2', 'usr/lib/libc.so.6', 'usr/lib/libm.so.6'] needed in files ['usr/bin/video-trimmer'])
video-trimmer W: Dependency gdk-pixbuf2 detected and implicitly satisfied (libraries ['usr/lib/libgdk_pixbuf-2.0.so.0'] needed in files ['usr/bin/video-trimmer'])
video-trimmer W: Dependency cairo detected and implicitly satisfied (libraries ['usr/lib/libcairo.so.2'] needed in files ['usr/bin/video-trimmer'])
video-trimmer W: Dependency gcc-libs detected and implicitly satisfied (libraries ['usr/lib/libgcc_s.so.1'] needed in files ['usr/bin/video-trimmer'])
video-trimmer W: Dependency pango detected and implicitly satisfied (libraries ['usr/lib/libpango-1.0.so.0'] needed in files ['usr/bin/video-trimmer'])
video-trimmer W: Dependency included, but may not be needed ('ffmpeg')
video-trimmer W: Dependency included, but may not be needed ('gst-plugins-good')
```

File: archlinuxcn/video-trimmer/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Maintainer: torculus <hidden> 
+# Contributor: Mark Wagie <mark dot wagie at proton dot me>
+pkgname=video-trimmer
+pkgver=0.8.2
+pkgrel=1
+pkgdesc="Trim videos quickly"
+arch=('x86_64')
+url="https://apps.gnome.org/VideoTrimmer"
+license=('GPL3')
```
Comment (Zenvie) [inline archlinuxcn/video-trimmer/PKGBUILD:9]:
```
video-trimmer E: GPL3 is not a valid SPDX license identifier. See https://spdx.org/licenses/ for valid identifiers, or prefix the identifier with 'LicenseRef-', if it is custom.
```

Comment (Kiri2002) [reply archlinuxcn/video-trimmer/PKGBUILD]: changed to : 'GPL-3.0-or-later'

File: archlinuxcn/video-trimmer/PKGBUILD (hunk 2 above)
Comment (Kiri2002) [reply archlinuxcn/video-trimmer/PKGBUILD]: Added the first 9 implicitly satisfied dependencies.
But ffmpeg and gst-plugins-good should be included refer to upstream'[ readme](https://gitlab.gnome.org/YaLTeR/video-trimmer/-/blob/master/README.md?ref_type=heads#format-support)

File: archlinuxcn/video-trimmer/PKGBUILD (hunk 1 above)
Comment (Kiri2002) [reply archlinuxcn/video-trimmer/PKGBUILD:48]: This is quit difficult and I couldn't find a good answer.

1. ELF lacks GNU_PROPERTY_X86_FEATURE_1_SHSTK:
[Detect missing GNU_PROPERTY_X86_FEATURE_1_SHSTK in ELF files.](https://gitlab.archlinux.org/pacman/namcap/-/issues/63)
[namcap reports GNU_PROPERTY_X86_FEATURE_1_SHSTK, what action to take?](https://bbs.archlinux.org/viewtopic.php?id=292194)

2. Unused shared library:
[Unused shared libraries](https://github.com/TokTok/c-toxcore/issues/359)

After reading those, still don't how to fix. I think need some helps.

Comment (Zenvie) [reply archlinuxcn/video-trimmer/PKGBUILD:41]: Please report this issue to the upstream.

Comment (Kiri2002) [reply archlinuxcn/video-trimmer/PKGBUILD:41]: > Please report this issue to the upstream.

reported [here](https://gitlab.gnome.org/YaLTeR/video-trimmer/-/issues/72)

**Pull request merged**